DOC-616 Update Serverless doc (#83)
redpanda-data/cloud-docs: modules/get-started/pages/cluster-types/serverless.adoc

====

== Get started with Serverless trial

New trials receive $100 (USD) in free credits to spend in the first 14 days. This should be enough to run Redpanda with reasonable throughput. No credit card is required. To continue using Serverless after your trial expires, you can enter a credit card and pay as you go. Any remaining credit balance is used before you are charged. You can view detailed billing activity for each cluster and edit payment methods on the *Billing* page. Each trial supports five Serverless clusters. 

When either the 100 credits expire or the 14 days in the trial expire, the clusters move into a suspended state, and you won't be able to access your data in either the Redpanda Cloud UI or with the Kafka API. There is a 7 day grace period following the end of the trial. After that, the data is permanently deleted. 

* To ask questions about the trial, use the *#serverless* https://redpandacommunity.slack.com/[Community Slack^] channel, or contact https://support.redpanda.com/hc/en-us/requests/new[support^].
* For information about billing after the trial ends, see xref:billing:billing.adoc[].

== Interact with your trial cluster

> After you click to start a trial, Redpanda instantly prepares an account for you. Your account includes a `welcome` cluster with a `hello-world` demo topic you can explore. The `hello-world` topic starts as a free read-only topic. It is materialized and becomes billable with your credits if you produce messages to it. 

Reviewer: ```suggestion
After you click to start a trial, Redpanda instantly prepares an account for you. in https://cloud.redpanda.com[Redpanda Cloud^]. Your account includes a `welcome` cluster with a `hello-world` demo topic you can explore. The `hello-world` topic starts as a free read-only topic. It is materialized and becomes billable with your credits if you produce messages to it. 
```

Author: Leaving out, because we have the link below + the UI opens from clicking to start trial.

== Interact with your trial cluster

After you click to start a trial, Redpanda instantly prepares an account for you. Your account includes a `welcome` cluster with a `hello-world` demo topic you can explore. The `hello-world` topic starts as a free read-only topic. It is materialized and becomes billable with your credits if you produce messages to it. 

== Interact with your cluster
Follow the steps in the UI to use `rpk` to interact with your cluster from the command line:

. Log in:
+
```
rpk cloud login
```

> Similar to Dedicated and BYOC clusters, you can interact with your Serverless cluster with `rpk` CLI commands or with the Redpanda Cloud UI. For example, you can run `rpk topic consume hello-world` to read the messages in that topic explaining the `rpk` steps to stream data. Here are some helpful `rpk` commands:

Reviewer: that's a good diff. We should avoid mentioning other offerings when we're in a guide about a specific offering. Unless we have a strong reason to.

In the https://cloud.redpanda.com[Redpanda Cloud UI^], you can navigate to the *Topics* page and open the `hello-world` topic to see the included messages. Under the *Actions* dropdown, you can produce messages to it. Add team members and grant them access with ACLs on the *Security* page. The *Overview* page lists your bootstrap server URL and security settings in the *How to connect - Kafka API* tab. 

> 

Reviewer: we should include a section about how to connect with your cluster here.
Either we give the steps here, or list where in the UI the user can get this information. 

Possibly move this text " The *Overview* page lists your bootstrap server URL and security settings in the *How to connect - Kafka API* tab. "  into the new heading

Here are some helpful `rpk` commands:

* xref:reference:rpk/rpk-cloud/rpk-cloud-login.adoc[`rpk cloud login`]: Use this to log in to Redpanda Cloud or to refresh the session.
* xref:reference:rpk/rpk-topic.adoc[`rpk topic`]: Use this to manage topics, produce data, and consume data. 
* xref:reference:rpk/rpk-profile/rpk-profile-print.adoc[`rpk profile print`]: Use this to view your `rpk` configuration and see the URL for your Serverless cluster.
* xref:reference:rpk/rpk-security/rpk-security-user.adoc[`rpk security user`]: Use this to manage users and permissions. 

NOTE: Redpanda Serverless is opinionated about Kafka configurations. For example, automatic topic creation is disabled. Some systems expect the Kafka service to automatically create topics when a message is produced to a topic that doesn't exist. Create topics on the *Topics* page or with `rpk topic create`.

> === Run a demo application

Reviewer: ```suggestion
=== Create a demo application
```

Reviewer: running might imply that we have an engine to run the languages. Create is more accurate

> To generate a sample application to connect with Redpanda, run xref:reference:rpk/rpk-generate/rpk-generate-app.adoc[`rpk generate app`]. Follow the commands in the terminal to run the application, create a demo topic, produce to the topic, and consume the data back.

Reviewer: I feel like we should say something to grab the attention of the users. Please modify to match our style guide
```suggestion
Redpanda can generate sample applications for you to interact with your cluster. Run xref:reference:rpk/rpk-generate/rpk-generate-app.adoc[`rpk generate app`] and pick your preferred programming language.  Follow the commands in the terminal to run the application, create a demo topic, produce to the topic, and consume the data back.
```

> == Create a Serverless cluster

Reviewer: ```suggestion
== Create a new Serverless cluster
```

To create additional Serverless clusters: 

> . Log in to https://cloud.redpanda.com[Redpanda Cloud^].

Reviewer: I wouldn't include this step. It's safe to assume that the user is logged in.

. On the **Clusters** page, click **Create cluster**, then click **Create Serverless cluster**. 
+

> Enter a cluster name, then select the resource group. If you don't have an existing resource group, you can create one.  

Reviewer: ```suggestion
Enter a cluster name, then select the resource group. If you don't have an existing resource group, you can create one and refresh this page.  
```

Reviewer: Pick which one makes more sense.

> Enter a cluster name, then select the resource group. If you don't have an existing resource group, you can create one.  

Reviewer: ```suggestion
Enter a cluster name, then select the resource group. If you don't have an existing resource group, you can create one and refresh this page.  
```

Reviewer: Pick which one makes more sense.

Author: edited the first option!
